libmamba/include/mamba/specs/channel.hpp

```diff
@@ -109,6 +109,7 @@ namespace mamba::specs
         auto clear_platforms() -> platform_list;
         void set_platforms(platform_list platforms);
 
+        [[nodiscard]] auto id() const -> const std::string&;
```

Comment:
Can we a add a doc saying what this is (a cross URL id) and its limitations (dependent on `channel_alias`).